src/main/java/org/team1540/robot2024/util/PhoenixTimeSyncSignalRefresher.java

```diff
@@ -0,0 +1,38 @@
+package org.team1540.robot2024.util;
+
+import com.ctre.phoenix6.BaseStatusSignal;
+import com.ctre.phoenix6.CANBus;
+import edu.wpi.first.wpilibj.Timer;
+import org.team1540.robot2024.Constants;
+
+import java.util.Arrays;
+
+import static org.team1540.robot2024.Constants.SwerveConfig.*;
+
+/**
+ * Helper class for refreshing CANivore TimeSynced status signals.
+ */
+public class PhoenixTimeSyncSignalRefresher {
+    private static BaseStatusSignal[] signals = new BaseStatusSignal[0];
+    private static final boolean isCANFD = CANBus.isNetworkFD(CAN_BUS);
+
+    private static double lastRefreshTimestamp = 0;
+
+    /**
+     * Registers status signals to be refreshed when {@link PhoenixTimeSyncSignalRefresher#refreshSignals()}is called
+     */
+    public static void registerSignals(BaseStatusSignal... newSignals) {
+        signals = Arrays.copyOf(signals, signals.length + newSignals.length);
+        for (int i = 0; i < newSignals.length; i++) signals[signals.length - 1 - i] = newSignals[i];
+    }
+
+    /**
+     * Refreshes all registered signals if not already done so in the current loop period
+     */
+    public static void refreshSignals() {
+        if (Timer.getFPGATimestamp() - lastRefreshTimestamp < Constants.LOOP_PERIOD_SECS) return;
+        lastRefreshTimestamp = Timer.getFPGATimestamp();
+        if (isCANFD) BaseStatusSignal.waitForAll(0.01, signals);
```

Comment:
Does this wait for them to send after calling, wait for an update since last time you checked, or prompt them to respond?

If the first or third (I can't find it in a cursory search of the phoenix docs), waiting up to half the loop time might be a little long

Reply:
It is the second one, however I can see how this can cause problems, especially since the signals only refresh every 0.01s, so you could still end up waiting 0.01s every time if your main robot loop and the status signal refresh loop are synced in just the right way (Status signals refresh just before `waitForAll()` is called, causing it to wait close to the full 0.01s before recieving an update). 

This is pretty easily fixed by just increasing odometry signals to 250hz. Alternatively, I could just reduce the timeout threshold here, but then we would sometimes use stale data (and would probably be better off not using timesync). An even more scuffed way to do this would to put all of the `waitForAll()`s on a separate thread like is done [here](https://github.com/Mechanical-Advantage/AdvantageKit/blob/main/example_projects/advanced_swerve_drive/src/main/java/frc/robot/subsystems/drive/PhoenixOdometryThread.java).